### PR TITLE
[server] Expose stats related to anomaly and feedback

### PR DESCRIPTION
**Pull request by shounakmk219**
The anomaly feedback related stats are exposed under AppAnalyticsApi as below:
```
{
  ...
  "anomalyStats": {
    "totalCount": 110,
    "countWithFeedback": 20,
    "feedbackStats": {
      "NO_FEEDBACK": 0,
      "ANOMALY": 15,
      "NOT_ANOMALY": 2,
      "ANOMALY_NEW_TREND": 0,
      "ANOMALY_EXPECTED": 3
    }
  }
}
```

### CODE REVIEW

File: thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/api/AnomalyStatsApi.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return feedbackStats;
+  }
+
+  public AnomalyStatsApi setFeedbackStats(
+      final Map<AnomalyFeedbackType, Long> feedbackStats) {
+    this.feedbackStats = feedbackStats;
+    return this;
+  }
+
+  public void incFeedbackStatCount(AnomalyFeedbackType type) {
```
Comment (suvodeep-pyne) [inline thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/api/AnomalyStatsApi.java:60]: Just use a setter here. Again, trying to stick to a convention where data pojos are simple pojos with no fancy operations.

File: thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java [hunk 2]
```diff
@@ -124,4 +127,19 @@ public ConfusionMatrix computeConfusionMatrixForAnomalies() {
...
+        .map(MergedAnomalyResultDTO::getFeedback)
+        .collect(Collectors.toList());
+  }
+
+  public AnomalyStatsApi computeAnomalyStats() {
+    List<AnomalyFeedback> feedbacks = getAnomalyFeedbacks();
+    AnomalyStatsApi stat = new AnomalyStatsApi()
+        .setTotalCount(anomalyManager.countParentAnomalies())
+        .setCountWithFeedback((long) feedbacks.size());
+    feedbacks.forEach(feedback -> stat.incFeedbackStatCount(feedback.getFeedbackType()));
```
Comment (suvodeep-pyne) [inline thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java:142]: I would just build the map here and set it on the POJO. so we don't have to even initialize it.

Motivation here is to keep business logic classes and data classes separate. so data classes like API classes are completely dumb! only job is to move around the resource classes and get serialized to json.

**Review by suvodeep-pyne — APPROVED** [review]
LGTM. minor comments. please address and merge.

File: thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java (hunk 2 above)
Comment (shounakmk219) [reply thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java]: Makes sense. Done.

File: thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/api/AnomalyStatsApi.java (hunk 1 above)
Comment (shounakmk219) [reply thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/api/AnomalyStatsApi.java]: Got it. Done.

**Pull request merged**